lib/logthrdest/logthrdestdrv.c

```diff
@@ -966,13 +967,26 @@ log_threaded_dest_driver_queue(LogPipe *s, LogMessage *msg,
 }
 
 static void
-_init_stats_key(LogThreadedDestDriver *self, StatsClusterKey *sc_key)
+_init_stats_legacy_key(LogThreadedDestDriver *self, StatsClusterKey *sc_key)
 {
   stats_cluster_logpipe_key_legacy_set(sc_key, self->stats_source | SCS_DESTINATION,
                                        self->super.super.id,
                                        self->format_stats_instance(self));
 }
 
+static void
+_init_stats_key(LogThreadedDestDriver *self, StatsClusterKey *sc_key)
+{
+  enum { labels_len = 1 };
+  static StatsClusterLabel labels[labels_len];
+
+  labels[0] = stats_cluster_label("id", self->super.super.id ? : "");
+  stats_cluster_logpipe_key_set(sc_key, "output_events_total", labels, labels_len);
+  stats_cluster_logpipe_key_add_legacy_alias(sc_key, self->stats_source | SCS_DESTINATION,
+                                             self->super.super.id,
+                                             self->format_stats_instance(self));
+}
+
```

Comment:
`_init_stats_key()` now provides the new-style "output_event_total" counter. It is passed to `log_queue_register_stats_counters()`:
```c
static void
_register_worker_stats(LogThreadedDestWorker *self)
{
  StatsClusterKey sc_key;

  stats_lock();
  _init_stats_key(self->owner, &sc_key);
  log_queue_register_stats_counters(self->queue, 0, &sc_key);
  stats_unlock();
}
```
It registers the common counters and also adds place for each queue implementation to register their own counters:
```c
void
log_queue_register_stats_counters(LogQueue *self, gint stats_level, const StatsClusterKey *sc_key)
{
  _register_common_counters(self, stats_level, sc_key);

  if (self->register_stats_counters)
    self->register_stats_counters(self, stats_level, sc_key);
}
```
Although it is not used by any of our queue implementation, but if it was, I think we should not always register the custom counters under the new `"output_event_total"` cluster.

I think we should pass both the new and the legacy keys, so the queue implementation can decide which one to use, or we should remove the possibility of registering custom counters altogether.

Reply:
I removed this abstract function. From the LogQueue's perspective, this could be a useful interface, but I don't think we will use it anytime soon.